ocean_provider/validation/RBAC.py

```diff
@@ -0,0 +1,146 @@
+#
+# Copyright 2021 Ocean Protocol Foundation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+import json
+from typing import Optional
+
+from ocean_lib.common.agreements.service_types import ServiceTypesIndices
+from ocean_lib.web3_internal.transactions import sign_hash
+
+from ocean_provider.exceptions import RequestNotFound
+from ocean_provider.utils.basics import get_provider_wallet
+from ocean_provider.utils.util import msg_hash
+
+
+class RBACValidator:
+    def __init__(
+        self,
+        request_name=None,
+        request=None,
+        payload: Optional[dict] = None,
+        assets: Optional[list] = None,
+        algorithms: Optional[list] = None,
+    ):
+        self.request = request
+        self.payload = payload if payload else dict()
+        if not request:
+            raise RequestNotFound("Request not found.")
+        action_mapping = self.get_action_mapping()
+        self.action = action_mapping[request_name]
+        self.credentials = {
+            "type": "address",
+            "address": get_provider_wallet().address,
+        }
+        self.component = "provider"
+        self.provider_address = get_provider_wallet().address
+        self.assets = assets if assets else []
+        self.algorithms = algorithms if algorithms else []
+        if request_name in ["ComputeRequest", "ComputeStartRequest"]:
```

Comment:
it seems we don't have an understanding of what belongs in the parent class, and what should be defined in each functiono specifically. if the request is one of those, please compute the additional_inputs variable inside the function, using self.payload which is generic to all requests.